Source/Core/DolphinQt/NetPlay/NetPlayDialog.cpp

```diff
@@ -458,6 +462,57 @@ void NetPlayDialog::show(std::string nickname, bool use_traversal)
   UpdateGUI();
 }
 
+void NetPlayDialog::UpdateDiscordPresence()
+{
+#ifdef USE_DISCORD_PRESENCE
+  // both m_current_game and m_player_count need to be set for the status to be displayed correctly
+  if (m_player_count == 0 || m_current_game.empty())
+    return;
+
+  const auto use_default = [this]() {
+    Discord::UpdateDiscordPresence(m_player_count, Discord::SecretType::Empty, "", m_current_game);
+  };
+
+  if (g_TraversalClient)
```

Comment:
This will be true even for clients which aren't the host. The other clients will have some other host code that isn't useful (this is due to how TraversalClient works), so this should also check `IsHosting()`. Besides, we probably don't want the clients accepting join requests on behalf of the host.